### PR TITLE
T/262: Implemented View#render method

### LINKED ISSUE #262: Should first call of view.element render view.template?
This might cause confusing situations.

Using `this.element` inside view constructor (eg to initialize `FocusTracker` or `KeystrokeHandler`) renders the template and makes it not possible to be extended.
```javascript
class MyView extends View {
    constructor( locale ) {
        super( locale );
         
        this.template = new Template( ... );

        this.focusTracker = new FocusTracker();
        this.fousTracker.add( this.element );
    }
}

const view = new MyView( locale );

Template.extend( view.template, { .... } );
```
And I see a warning:
```
template-extend-render: Attempting to extend a template which has already been rendered.
```
but my view is not even initialized.

To workaround this problem I can move `FT` init to the view `init()` but it took me a while to figure out why I can't extend this template.

### LINKED ISSUE #302: [Tests] Broken image toolbar manual test
http://localhost:8125/ckeditor5-ui/tests/manual/imagetoolbar/imagetoolbar.html shows errors and won't start.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Implemented `View#render` method which replaces the `#element` rendering upon the first reference and incorporates the `#init` method functionality. Closes #262. Closes #302.

From now on `View#setTemplate` and `View#extendTemplate` methods are recommended as a shorthand for `view.template = new Template( { ... } )` and `Template.extend( view.template )`.

BREAKING CHANGE: The `init()` method in various UI components has been renamed to `render()`. Please refer to the documentation to learn more.

BREAKING CHANGE: The `View#element` is no longer a getter which renders an element when first referenced. Use the `View#render()` method instead.

BREAKING CHANGE: `Template#children` property became an `Array` (previously `ViewCollection`).

BREAKING CHANGE: `View#addChildren` and `View#removeChildren` methods became `#registerChildren` and `#deregisterChildren`.

BREAKING CHANGE:  The DOM structure of the `StickyPanelView` has changed along with the class names. Please refer to the component documentation to learn more.

---

### Additional information

This PR comes with a branch constellation centralized in https://github.com/ckeditor/ckeditor5/tree/t/ckeditor5-ui/262:

* https://github.com/ckeditor/ckeditor5-core/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-editor-balloon/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-editor-classic/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-editor-inline/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-heading/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-image/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-link/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-theme-lark/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-utils/tree/t/ckeditor5-ui/262
* https://github.com/ckeditor/ckeditor5-upload/tree/t/ckeditor5-ui/262

Merge commit message for related branches:

```
Internal: Aligned the UI to the latest API of the framework (see ckeditor/ckeditor5-ui#262).
```

#### Misc

* Removed obsolete image toolbar test (https://github.com/ckeditor/ckeditor5-ui/issues/302), which has been handled by the `ContextualBalloon` for a while.

### CODE REVIEW

File: src/iframe/iframeview.js [hunk 1]
```diff
@@ -38,45 +37,27 @@ export default class IframeView extends View {
...
-		return this._iframePromise;
+		this.on( 'loaded', () => {
+			deferred.resolve();
+		} );
+
+		super.render();
+
+		return new Promise( ( resolve, reject ) => {
+			deferred = { resolve, reject };
+		} );
```
Comment (oskarwrobel) [inline src/iframe/iframeview.js]: Why not just?
```javascript
render() {
    return new Promise( resolve => {
        this.on( 'loaded', resolve );
        super.render();
    } ) ;
}
```

**Comment by oskarwrobel** [comment]
It's worth to mention about `View#setTemplate` and `View#extendTemplate` methods in comment message.

**Comment by oskarwrobel** [comment]
https://github.com/ckeditor/ckeditor5-utils/compare/t/ckeditor5-ui/262?expand=1#diff-b7a0ddd6b2b22f178c9c2e5bbf8fac34R7 - should be `isdomnode`

File: src/iframe/iframeview.js (hunk 1 above)
Comment (oleq) [reply src/iframe/iframeview.js]: If tests are passing, I think it is the right way to render an `IframeView`

Comment (oleq) [reply src/iframe/iframeview.js]: I meant: you're probably right 😋

Comment (oskarwrobel) [reply src/iframe/iframeview.js]: It works fine but I think we don't need to resolve this promise outside of the promise scope. `deferred` variable doesn't have too much sense here. But it's a detail :)

**Comment by oleq** [comment]
FYI: I've just pushed the fixes to the branches.

**Pull request merged**